web/resources.go

```diff
@@ -235,6 +235,37 @@ func getIRS() []dao.RunningService {
 	return services
 }
 
+func restGetServicesForMigration(w *rest.ResponseWriter, r *rest.Request, client *node.ControlClient) {
+	serviceID, err := url.QueryUnescape(r.PathParam("serviceId"))
+	if err != nil {
+		restBadRequest(w, err)
+		return
+	}
+	services := []service.Service{}
+	if err := client.GetServiceList(serviceID, &services); err != nil {
+		glog.Errorf("Could not get services: %v", err)
+		restServerError(w, err)
+		return
+	}
+	w.WriteJson(&services)
+}
+
+func restPostServicesForMigration(w *rest.ResponseWriter, r *rest.Request, client *node.ControlClient) {
+	var smr dao.ServiceMigrationRequest
+	err := r.DecodeJsonPayload(&smr)
+	if err != nil {
+		glog.Errorf("Could not decode services for migration: %v", err)
+		restServerError(w, err)
```

Comment:
Just an FYI - this is pattern (used throughout serviced) will send a RESTafarian to screaming. The proper HTTP response to malformed JSON is 400, not 500. No need to change in this case, but it's a bigger change that should be made throughout the serviced REST API.